core/connection/manager_test.go

```diff
@@ -431,14 +517,13 @@ func (mpm *MockPaymentIssuer) Stop() {
 	close(mpm.stopChan)
 }
 
-func newMockStatusSender() *mockStatusSender {
-	return &mockStatusSender{}
-}
-
 type mockStatusSender struct {
 	sentMsg *connectivity.StatusMessage
+	sync.Mutex
 }
 
-func (s mockStatusSender) Send(dialog communication.Sender, msg *connectivity.StatusMessage) {
+func (s *mockStatusSender) Send(dialog communication.Sender, msg *connectivity.StatusMessage) {
```

Comment:
you might want to implement a getter under lock here, as races can still happen if you directly try to access the sentMsg variable.

Reply:
sentMsg is accessed in unit tests asserts which are running in a single thread so there is no way for race condition to happen. If it happens test will fail which will indicate that something is wrong with unit tests logic.